server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -88,11 +88,13 @@ public IndexerSQLMetadataStorageCoordinator(
             final ResultIterator<byte[]> dbSegments =
                 handle.createQuery(
                     String.format(
-                        "SELECT payload FROM %s WHERE used = true AND dataSource = :dataSource",
+                        "SELECT payload FROM %s WHERE used = true AND dataSource = :dataSource AND start >= :start and \"end\" <= :end  AND used = true",
```

Comment:
what about segments that partially overlap the interval ?
earlier they were also being returned as usedSegments, I think we need to preserve that behaviour.